pyanaconda/network.py

```diff
@@ -1151,14 +1242,17 @@ def write_hostname(rootpath, ksdata, overwrite=False):
 
 def disableIPV6(rootpath):
     cfgfile = os.path.normpath(rootpath + ipv6ConfFile)
-    if ('noipv6' in flags.cmdline
-        and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
-                for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
-        log.info('Disabling ipv6 on target system')
-        with open(cfgfile, "a") as f:
-            f.write("# Anaconda disabling ipv6 (noipv6 option)\n")
-            f.write("net.ipv6.conf.all.disable_ipv6=1\n")
-            f.write("net.ipv6.conf.default.disable_ipv6=1\n")
+    try:
+        if ('noipv6' in flags.cmdline
+            and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
+                    for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
```

Comment:
I don't want to create another function with just one caller, but yes, the code is ugly and may be unreadable, but above all,  due to the usage of "all" it requires the exception block to include the stuff it shouldn't so I will rewrite it.